app/components/modalComponent/teacher/deleteTeacherModal.controller.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Created by maddog on 2/28/2018.
+ */
+
+(
+    function(){
+        'use strict'
+
+        angular.module('MockApp')
+            .controller('DeleteTeacherModalController',DeleteTeacherModalController);
+
+        DeleteTeacherModalController.$inject=['$uibModalInstance','TeacherService','Teacher'];
+
+        function DeleteTeacherModalController($uibModalInstance,TeacherService,Teacher){
+            var vm = this;
+
+            vm.teacherName = Teacher.name;
```

Comment:
What happens if Teacher is undefined or null here? We may need to safe check this variable